onnxruntime/core/mlas/lib/amd64/QgemmU8U8KernelAvx2.asm

```diff
@@ -813,281 +767,59 @@ ComputeBlock MACRO ColumnCount, RowCount, VectorOffset, BroadcastOffset
-        sub     rsi,2
+        sub     rsi,2*4
         jb      ProcessRemainingBlocks
 
-ComputeBlockLoop:
+ComputeBlockBy2Loop:
         ComputeBlock ColumnCount, RowCount, 0, 0
         ComputeBlock ColumnCount, RowCount, 32, 4
         add     rcx,2*4                     ; advance matrix A by 2 pairs
 IF RowCount GT 3
         add     rbx,2*4                     ; advance matrix A plus 3 rows by 2 pairs
 ENDIF
-        add     rdx,2*32                    ; advance matrix B by 64 columns
-        sub     rsi,2                       ; subtract pairs remaining
-        jae     ComputeBlockLoop
+        add     rdx,2*32                    ; advance matrix B
+        sub     rsi,2*4
+        jae     ComputeBlockBy2Loop
 
 ProcessRemainingBlocks:
-        add     rsi,2                       ; correct for over-subtract above
+        add     rsi,2*4                     ; correct for over-subtract above
         jz      ComputeBlockLoopExit
         ComputeBlock ColumnCount, RowCount, 0, 0
-        add     rdx,32                      ; advance matrix B by 32 columns
+        add     rdx,32                      ; advance matrix B
 ELSE
-ComputeBlockLoop:
+ComputeBlockBy1Loop:
         ComputeBlock ColumnCount, RowCount, 0, 0
         add     rcx,4                       ; advance matrix A by 1 pair
 IF RowCount GT 3
         add     rbx,4                       ; advance matrix A plus 3 rows by 1 pair
 ENDIF
-        add     rdx,32
-        dec     rsi                         ; decrement pairs remaining
-        jnz     ComputeBlockLoop
+        add     rdx,32                      ; advance matrix B
+        sub     rsi,4
+        jnz     ComputeBlockBy1Loop
 ENDIF
 
 ComputeBlockLoopExit:
-IF RowCount GT 3
-        lea     rbx,[r8+rax*2]              ; compute matrix C plus 3 rows
-        add     rbx,rax
-ENDIF
-
-        ENDM
-
-;
-; Macro Description:
-;
-;   This macro generates code to compute matrix multiplication for a fixed set
-;   of rows.
-;
-; Arguments:
-;
-;   RowCount - Supplies the number of rows to process.
-;
-;   Fallthrough - Supplies a non-blank value if the macro may fall through to
-;       the ExitKernel label.
-;
-; Implicit Arguments:
-;
-;   rax - Supplies the length in bytes of a row from matrix C.
-;
-;   rcx - Supplies the address of matrix A.
-;
-;   rdx - Supplies the address of matrix B.
-;
-;   r8 - Supplies the address of matrix C.
-;
-;   rdi - Supplies the address of matrix A.
-;
-;   rbp - Supplies the number of columns from matrix B and matrix C to iterate
-;       over.
-;
-;   r9 - Supplies the number of paired columns from matrix A and the number of
-;       paired rows from matrix B to iterate over.
-;
-;   r10 - Supplies the length in bytes of a row from matrix A.
-;
-;   r12 - Supplies the address of the row sum vector.
-;
-;   r13 - Supplies the address of the column sum vector.
-;
-;   r14b - Supplies the zero mode flag.
-;
-
-ProcessCountM MACRO RowCount, Fallthrough
```

Comment:
It would be nicer to put these procedures like ProcessCountM/ComputeBlockLoop to some common files instead of distributed across  different files for readability.